src/Core/FlatpakBackend.vala

```diff
@@ -262,6 +262,42 @@ public class AppCenterCore.FlatpakBackend : Backend, Object {
         return installed_apps;
     }
 
+    public Gee.Collection<Package> get_packages_by_release_date () {
+        var apps = new Gee.TreeSet<AppCenterCore.Package> (compare_packages_by_release_date);
+
+        user_appstream_pool.get_components ().foreach ((comp) => {
+            var packages = get_packages_for_component_id (comp.get_id ());
+            apps.add_all (packages);
+        });
+
+        system_appstream_pool.get_components ().foreach ((comp) => {
+            var packages = get_packages_for_component_id (comp.get_id ());
+            apps.add_all (packages);
+        });
```

Comment:
```suggestion
    public Gee.Collection<Package> get_native_packages_by_release_date () {
        var apps = new Gee.TreeSet<AppCenterCore.Package> (compare_packages_by_release_date);

        user_appstream_pool.get_components ().foreach ((comp) => {
            var packages = get_packages_for_component_id (comp.get_id ());
            foreach (var package in packages) {
                if (package.is_native) {
                    apps.add (package);
                }
            }
        });

        system_appstream_pool.get_components ().foreach ((comp) => {
            var packages = get_packages_for_component_id (comp.get_id ());
            foreach (var package in packages) {
                if (package.is_native) {
                    apps.add (package);
                }
            }
        });
```

Since you only really want the native packages from this method anyway, I'd be tempted to rename and adapt it to do exactly that

Reply:
Makes sense to me :+1: